xds/src/main/java/io/grpc/xds/Filter.java

```diff
@@ -32,7 +32,7 @@
  * {@link ClientInterceptorBuilder} or {@link ServerInterceptorBuilder} or both, indicating it is
  * capable of working on the client side or server side or both, respectively.
  */
-interface Filter {
+public interface Filter {
```

Comment:
Again this change is needed because of `FilterChainMatchingHandler` being in a different package. Also this is the case of `@VisibleForTesting`

Reply:
it's true it is more widely visible than necessary just for test now.
But I thought usually @visibleForTesting is for package level visibility not if its already public. Soon it will be used for non test routing configs.